src/data/markdown/docs/01 guides/05 Testing Guides/04 Large-scale k6 testing.md

```diff
@@ -0,0 +1,484 @@
+```
+
+</div> 
+
+To fix this issue your checks must be resilient to any response type. This change will fix the above problem.
+
+<div class="code-group" data-props='{"labels": ["resilient check"]}'>
+
+```javascript
+let checkRes = check(res, {
+    "Homepage body size is 11026 bytes": (r) => r.body && r.body.length === 11026
+});
+```
+
+</div>
+
+### Monitor the load-generator server
+
+If you are running a test for the first time, it's a good idea to keep an eye on the available resources while the test is running.
+The easiest way to do so is to ssh to the server with 3 sessions:
+1. To run k6 
+2. To monitor CPU and memory
+3. To monitor the network
+
+For monitoring CPU and memory we recommend [htop](https://en.wikipedia.org/wiki/Htop). 
+For monitoring network, we recommend [iftop](https://en.wikipedia.org/wiki/Iftop)
+
+Here's a screenshot of 3 terminal sessions showing k6, iftop and htop.
+![k6 iftop htop](./images/large-scale-testing-3-terminals.png)
+
+
+## Additional k6 flags to achieve better performance
+
+### --compatibility-mode=base
+
+If you are pushing the limits of the hardware, this is the most impactful k6 setting you can enable.
+
+This setting disables the internal babel transpilation from ES6 to ES5 and inclusion of corejs library.
+
+<div class="doc-blockquote" data-props='{}'>
+
+> ### Some background
+> k6 at its core executes ECMAScript 5.1 code. Most k6 script examples and documentation is written in ECMAScript 6+. 
+> When k6 gets ES6+ code as input, it transpiles it to ES5.1 using babel and loads corejs to enable commonly used APIs.
+> This works very well for 99% of use cases, but it adds significant overheard with large tests.
+
+</div>
+
+To get the best performance out of k6, it's best to transpile the scripts outside of k6 using webpack.
+
+In [k6-hardware-benchmark](https://github.com/loadimpact/k6-hardware-benchmark) repository, we have prepared an efficient transpilation scheme that produces performant ES5.1 code for k6. 
+
+Use it like this:
+
+<div class="code-group">
+
+```bash
+git clone https://github.com/loadimpact/k6-hardware-benchmark/
+cd k6-hardware-benchmark
+yarn install
+yarn run to-es5 someplace/yourscript.js
+# your ES5 script is in someplace/yourscript.es5.js
+```
+</div>
+
+
+Once your code is transpiled, run it like this:
+
+<div class="code-group">
+
+```bash
+k6 run -o cloud --compatibility-mode=base someplace/yourscript.es5.js
+```
+</div>
+
+k6 will use about 50% of memory in comparison to running the original script. It will also reduce the CPU load. 
+
+### discardResponseBodies
+
+You can tell k6 to not process the body of the response by setting `discardResponseBodies` in the options object like this:
+
+<div class="code-group">
+
+```js
+export let options = {
+	discardResponseBodies: true,
+};
+```
+</div>
+
+k6 by default loads the response body of the request into memory. This causes much higher memory consumption, and often is completely unnecessary.
```

Comment:
You can add "but if you need it for some requests you can use [Params.responseType](https://k6.io/docs/javascript-api/k6-http/params-k6-http)" , but with an example